protocols/ping/src/protocol.rs

```diff
@@ -75,143 +94,126 @@ where
+where
+    TSocket: AsyncRead + AsyncWrite,
 {
     type Item = Duration;
     type Error = io::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
-            let new_state = match self.inner {
-                PingDialerInner::Write { ref mut inner } => {
-                    let (socket, ping_payload) = try_ready!(inner.poll());
-                    PingDialerInner::Flush {
-                        inner: tokio_io::io::flush(socket),
-                        ping_payload,
+            self.state = match self.state {
+                PingDialerState::Write { ref mut inner } => {
+                    let (socket, payload) = try_ready!(inner.poll());
+                    PingDialerState::Flush {
+                        inner: nio::flush(socket),
+                        payload,
                     }
                 },
-                PingDialerInner::Flush { ref mut inner, ping_payload } => {
+                PingDialerState::Flush { ref mut inner, payload } => {
                     let socket = try_ready!(inner.poll());
                     let started = Instant::now();
-                    PingDialerInner::Read {
-                        inner: tokio_io::io::read_exact(socket, [0; 32]),
-                        ping_payload,
+                    PingDialerState::Read {
+                        inner: nio::read_exact(socket, [0; 32]),
+                        payload,
                         started,
                     }
                 },
-                PingDialerInner::Read { ref mut inner, ping_payload, started } => {
-                    let (socket, obtained) = try_ready!(inner.poll());
-                    let ping_time = started.elapsed();
-                    if obtained != ping_payload {
-                        return Err(io::Error::new(io::ErrorKind::InvalidData,
-                            "Received ping payload doesn't match expected"));
+                PingDialerState::Read { ref mut inner, payload, started } => {
+                    let (socket, payload_received) = try_ready!(inner.poll());
+                    let rtt = started.elapsed();
+                    if payload_received != payload {
+                        return Err(io::Error::new(
+                            io::ErrorKind::InvalidData, "Ping payload mismatch"));
                     }
-                    PingDialerInner::Shutdown {
-                        inner: tokio_io::io::shutdown(socket),
-                        ping_time,
+                    PingDialerState::Shutdown {
+                        inner: nio::shutdown(socket),
+                        rtt,
                     }
                 },
-                PingDialerInner::Shutdown { ref mut inner, ping_time } => {
-                    let _ = try_ready!(inner.poll());
-                    return Ok(Async::Ready(ping_time));
+                PingDialerState::Shutdown { ref mut inner, rtt } => {
+                    try_ready!(inner.poll());
```

Comment:
Does that not produce a warning? The `let _ = ` are usually here to silence warnings about unused results.

Reply:
It does not produce a warning in this case.